docs/gh-pages-build.js

```diff
@@ -0,0 +1,142 @@
+#!/usr/local/bin/node
+/**
+ * Build a release for the gh-pages docs site.
+ */
+
+var fs = require('fs');
+var execSync = require('child_process').execSync;
+
+var usage = '\nbuild <vn.n.n[-pre[.n]] | HEAD> [-p]\n';
+var versionsFile = './src/www/versions.json';
```

Comment:
Well, much of the git wrangling assumes we're in docs, and it would get super messy and hard-to-read if we `path.resolve` all of those.

I was assuming this is going to be called in an automated fashion either for HEAD builds, or as part of @hai-cea's release process, or, if run manually, by using the `npm start` script. 

No harm in making it fool proof, but much easier to `process.chdir('__dirname');` though I think?